test_plans/spec-constants.asciidoc

```diff
@@ -0,0 +1,562 @@
+    `SYCL_EXTERNAL`.
+  - The separate translation unit has an external declaration for the
+    `specialization_id` variable, which is defined in the first translation
+    unit.
+
+==== Pass kernel handler object by value to a `SYCL_EXTERNAL` function
+
+* This test runs only if the implementation defines `SYCL_EXTERNAL`.
+* Same test as in <<sec:external-handler-by-ref>>, except pass the
+  `kernel_handler` object by value.
+
+[[sec:multiple]]
+=== Multiple spec constants
+
+* Declare several `specialization_id` variables of the tested type in the
+  global namespace.  All the default values are different and none are zero.
+* Create a `queue` from the tested device and call `queue::submit()`.
+* Set the values of some of the spec constants via
+  `handler::set_specialization_constant()` but do not set the values for all of
+  them.
+* Submit a kernel via `handler::single_task()`.
+* Read the values of the spec constant via
+  `kernel_handler::get_specialization_constant()` and make sure we get the
+  expected value from each (either the value we set or the default value).
+
+=== Multiple spec constants with `kernel_bundle`
+
+* Same test as in <<sec:multiple>>, except set spec constants in a
+  `kernel_bundle`, build the bundle and register the bundle with a handler.
+
+[[sec:two-command]]
+=== Two command groups that read the same spec constant, both set value
+
+* Declare a `specialization_id` variable in the global namespace for the
+  tested type.  The variable's default value has some non-zero value.
+* Declare a single object whose type is a class with `operator(handler &)`.
+  This object will be the command group handler for our test.
+* Create a `queue` from the tested device and call `queue::submit()` twice.
+  Each call passes the same command group handler object described above.
+* Each command group handler sets the value of the spec constant to a
+  different value.
+* Each command group handler submits a kernel via `handler::single_task()`.
+* Each kernel reads the spec constant via
+  `kernel_handler::get_specialization_constant()`.
+* Verify that the value read in the kernel is the same as the value set in
+  the command group handler which launched that kernel instance.
+* The two kernels should run in parallel for this test.
+
+=== Two command groups that read the same spec constant, only one sets value
+
+Same test as in <<sec:two-command>> except only one of the command group
+handlers sets the value of the spec constant.  The kernel instance that is
+launched from the handler that does not set a value should read the spec
+constant's default value.
+
+[[sec:declarations]]
+=== Spec constant defined in various ways
+
+Do the following test for a `specialization_id` variable defined in the
+following ways:
+
+* Defined in a non-global namespace.
+* Defined in the global namespace as `const`.
+* Defined in the global namespace as `constexpr`.
+* Defined in the global namespace as `inline`.
```

Comment:
This is probably illegal now, looking at KhronosGroup/SYCL-Docs#133

Reply:
Fixed in da04e6d5eef9f5c3f9bc0f803ab656a38e3e4744